src/code/audio/audio_seqplayer.c

```diff
@@ -1,73 +1,2314 @@
+}
+
+s32 AudioSeq_SeqLayerProcessScriptStep2(SequenceLayer* layer) {
+    SequenceChannel* channel = layer->channel;
+    SeqScriptState* state = &layer->scriptState;
+    SequencePlayer* seqPlayer = channel->seqPlayer;
+    u8 cmd;
+    u8 cmdArg8;
+    u16 cmdArg16;
+    u16 velocity;
+
+    if (D_801E05B4) {}
+
+    for (;;) {
+        cmd = AudioSeq_ScriptReadU8(state);
+
+        // Note Commands
+        // To be processed in AudioSeq_SeqLayerProcessScriptStep3
+        if (cmd <= 0xC0) {
+            return cmd;
+        }
+
+        // Control Flow Commands
+        if (cmd >= 0xF2) {
+            cmdArg16 = AudioSeq_GetScriptControlFlowArgument(state, cmd);
+
+            if (AudioSeq_HandleScriptFlowControl(seqPlayer, state, cmd, cmdArg16) == 0) {
+                continue;
+            }
+            AudioSeq_SeqLayerDisable(layer);
+            return -1;
+        }
+
+        switch (cmd) {
+            case 0xC1: // layer: set short note velocity
+            case 0xCA: // layer: set pan
+                cmdArg8 = *(state->pc++);
+
+                if (cmd == 0xC1) {
+                    layer->velocitySquare = SQ(cmdArg8) / SQ(127.0f);
+                } else {
+                    layer->pan = cmdArg8;
+                }
+                break;
+
+            case 0xC9: // layer: set short note gatetime
+            case 0xC2: // layer: set transposition in semitones
+                cmdArg8 = *(state->pc++);
+                if (cmd == 0xC9) {
+                    layer->gateTime = cmdArg8;
+                } else {
+                    layer->transposition = cmdArg8;
+                }
+                break;
+
+            case 0xC4: // layer: continuous notes on
+            case 0xC5: // layer: continuous notes off
+                if (cmd == 0xC4) {
+                    layer->continuousNotes = true;
+                } else {
+                    layer->continuousNotes = false;
+                }
+                layer->bit1 = false;
+                AudioPlayback_SeqLayerNoteDecay(layer);
+                break;
+
+            case 0xC3: // layer: set short note default delay
+                cmdArg16 = AudioSeq_ScriptReadCompressedU16(state);
+                layer->shortNoteDefaultDelay = cmdArg16;
+                break;
+
+            case 0xC6: // layer: set instrument
+                cmd = AudioSeq_ScriptReadU8(state);
+                if (cmd >= 0x7E) {
+                    if (cmd == 0x7E) {
+                        // Sfxs
+                        layer->instOrWave = 1;
+                    } else if (cmd == 0x7F) {
+                        // Drums
+                        layer->instOrWave = 0;
+                    } else {
+                        // Synthetic Wave
+                        layer->instOrWave = cmd;
+                        layer->instrument = NULL;
+                    }
+
+                    if (cmd == 0xFF) {
+                        layer->adsr.decayIndex = 0;
+                    }
```

Comment:
So... command `0xC6 0xFF` resets decay? But why... we already have a set decay index command..